File: LeetCode/Easy/0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int sz = nums.size();
        bool isFin = false;
        vector<int> ans(2);
        for (int i=0; i<sz; i++){
            for (int j=i+1; j<sz; j++){
                if (nums[i] + nums[j] == target){
                    ans[0] = i, ans[1] = j;
                    isFin = true;
                    break;
                }
            }
            if (isFin)
                break;
        }
        return ans;
    }
};

File: LeetCode/Easy/0001-two-sum/0001-two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        Map<Integer, Integer> visited = new HashMap<>();

        for (int i=0; i<n; i++){
            int num = nums[i];
            int t = target - num;

            if (visited.containsKey(t)){
                return new int[]{visited.get(t), i};
            }
            else {
                visited.put(num, i);
            }
        }

        return new int[2];
    }
}